### DISCUSSION TITLE
Towards support for i.MX8X

### DISCUSSION THREAD

**Discussion by saranli** [Freescale/libimxvpuapi#58] [General]
I was wondering what it would take to add support for i.MX8X platform's VPU encoder? Despite my lack of experience in the VPU area specifically, due to the possibility of using this platform for our robotics projects, I might be willing to try getting things to work if any of the experienced developers here can provide a rough roadmap. This is unless, of course, there are specific reasons why this would be a waste of time (e.g. i.MX8X not being sufficiently commonly used etc.) that I am not aware of.

Best regards.
- Uluç Saranlı

**Comment by otavio** [comment]
We are certainly open to add it. No reason to not include it.

**Reply by saranli** [reply]
Thank you for the speedy reply. If you could give me a sense of whether it's a reasonable or huge undertaking, as well as possibly a few pointers on how you would proceed exploring it, those would be good starting points for me. So, I am essentially looking for informed guidance based on your experience.
Thanks!

**Comment by dv1** [comment]
Are you using Yocto? If so, what version?

**Reply by saranli** [reply]
Yes, we are using Digi Embedded Yocto 3.2 on a ConnectCore 8X SBC Pro board. (https://www.digi.com/products/embedded-systems/digi-connectcore/single-board-computers/digi-connectcore-8x-sbc-pro). They are expected to have 4.0 support later this year I guess,, but it's not out yet. I am guessing I could replace various components (i.e. gstreamer etc.) with newer, more up-to-date versions if that's what it would take to get things going.

**Comment by saranli** [comment]
There seems to be some activity in some of the branches regarding this issue, could you give a brief update on where things are? Is the encoder implementation usable in a reliable fashion? Since my last comment, the board received Yocto 4.0 support. so the rest of the Yocto image will be up to date. I think I can start some testing on the board if any of the branches are usable.
